elastic/kibana
# [Security Solution][Exceptions] - Fix add exception to rule using data view from alert

## Summary

Addresses https://github.com/elastic/kibana/issues/138231

There's a bit of a race condition happening between data views/index patterns. When a rule has a data view defined, we really shouldn't be making calls to fetch the default index patterns which is what had started happening here, however then once the data view id was defined, the fetch index was cut short but the `useFetchIndex` hook never reset to loading `false`, hence the perpetual loading state. That initial call should be avoided all together so that's what this change does. There's still a little funkiness in the code here that I'd like to look at during our work in 8.5.

## Test

- Create a rule using a data view
- Wait for rule to generate alerts
- Select to add exception from the rule details alerts table
- Add exception flyout should load as expected

### Checklist
